test/src/unittests/tonal/test_pitchsaliencefunction.py

```diff
@@ -0,0 +1,223 @@
+        mags = [1, -1, 1] # length 3
+        self.assertRaises(EssentiaException, lambda: PitchSalienceFunction()(freqs, mags))
+
+    def testRegressionTest(self):
+        filename = join(testdata.audio_dir, 'recorded', 'vignesh.wav')
+        audio = MonoLoader(filename=filename, sampleRate=44100)()
+
+        hopSize = 128
+        frameSize = 2048
+        sampleRate = 44100
+        guessUnvoiced = True
+
+        run_windowing = Windowing(type='hann', zeroPadding=3*frameSize) # Hann window with x4 zero padding
+        run_spectrum = Spectrum(size=frameSize * 4)
+        run_spectral_peaks = SpectralPeaks(minFrequency=1,
+                                           maxFrequency=20000,
+                                           maxPeaks=100,
+                                           sampleRate=sampleRate,
+                                           magnitudeThreshold=0,
+                                           orderBy="magnitude")
+        run_pitch_salience_function = PitchSalienceFunction()
+        run_pitch_salience_function_peaks = PitchSalienceFunctionPeaks()
+        run_pitch_contours = PitchContours(hopSize=hopSize)
+        run_pitch_contours_melody = PitchContoursMelody(guessUnvoiced=guessUnvoiced,
+                                                        hopSize=hopSize)
+        run_predom_pitch_melodia = PredominantPitchMelodia()
+
+        # Now we are ready to start processing.
+        # 1. pass it through the equal-loudness filter
+        audio = EqualLoudness()(audio)
+
+        salience_peaks_bins_array = []
+        salience_peaks_saliences_array = []
+
+        # 2. Cut audio into frames and compute for each frame:
+        #    spectrum -> spectral peaks -> pitch salience function -> pitch salience function peaks
+        for frame in FrameGenerator(audio, frameSize=frameSize, hopSize=hopSize):
+            frame = run_windowing(frame)
+            spectrum = run_spectrum(frame)
+            peak_frequencies, peak_magnitudes = run_spectral_peaks(spectrum)
+            salience = run_pitch_salience_function(peak_frequencies, peak_magnitudes)
+            salience_peaks_bins, salience_peaks_saliences = run_pitch_salience_function_peaks(salience)
+            salience_peaks_bins_array.append(salience_peaks_bins)
+            salience_peaks_saliences_array.append(salience_peaks_saliences)
+
+        bins, saliences, startTimes, duration = run_pitch_contours(salience_peaks_bins_array, salience_peaks_saliences_array)
+        pitch1, pitchConfidence1 = run_pitch_contours_melody(bins, saliences, startTimes, duration) 
+        pitch2, pitchConfidence2 = run_predom_pitch_melodia(audio)
+
+        self.assertAlmostEqualVectorFixedPrecision(pitch1[:600], pitch2[:600], 8)
+        #TODO why do the outputs align up to a certain array point?
+        self.assertAlmostEqualVectorFixedPrecision(pitch1[:721], pitch2[:721], 8)
+        #self.assertAlmostEqualVectorFixedPrecision(pitchConfidence1, pitchConfidence2, 8)        
```

Comment:
This test is running PitchContour* algorithms, however this not what are expected to test here. We want to have a regression test validating that the values of the PitchSalienceFunction output are correct.

The type of test implemented here should go to the algorithms related to tracking pitch contours.

Reply:
then this need npy files

Reply:
The philosophy was to create an end to end test, that would YES involve other algorithms, but would have the advantage of not needing NPY files.